tests/Hedgehog.CSharp.Tests/Hedgehog.CSharp.Tests/LinqTests.cs

```diff
@@ -0,0 +1,126 @@
+        }
+
+        [Fact]
+        public void CanSelectFromTwoWithAssertion()
+        {
+            Check(
+                from x in ForAll(Gen.Bool)
+                from y in ForAll(Gen.Bool)
+                select Assert.True((x || !x) && (y || !y)));
+        }
+
+        [Fact]
+        public void CanSelectFromTwoWithBool()
+        {
+            Check(
+                from x in ForAll(Gen.Bool)
+                from y in ForAll(Gen.Bool)
+                select (x || !x) && (y || !y));
+        }
+
+        [Fact]
+        public void CanSelectFromThreeWithAssertion()
+        {
+            Check(
+                from x in ForAll(Gen.Bool)
+                from y in ForAll(Gen.Bool)
+                from z in ForAll(Gen.Bool)
+                select Assert.True(x || y || z || (!x || !y || !z)));
+        }
+
+        [Fact]
+        public void CanSelectFromThreeWithBool()
+        {
+            Check(
+                from x in ForAll(Gen.Bool)
+                from y in ForAll(Gen.Bool)
+                from z in ForAll(Gen.Bool)
+                select x || y || z || (!x || !y || !z));
+        }
+
+        [Fact]
+        public void CanUseWhereWithAssertion()
+        {
+            Check(20,
+                from x in ForAll(Gen.Bool)
+                where x == true
+                from y in ForAll(Gen.Bool)
+                where y == false
+                select Assert.True(x && !y));
+        }
+
+        [Fact]
+        public void CanUseWhereWithBool()
+        {
+            Check(20,
+                from x in ForAll(Gen.Bool)
+                where x == true
+                from y in ForAll(Gen.Bool)
+                where y == false
+                select x && !y);
+        }
+
+        [Fact]
+        public void CanDependOnEarlierValuesWithAssertion()
+        {
+            Check(
+                from i in ForAll(Gen.Int32(Range.Constant(1, 10)))
+                from j in ForAll(Gen.Int32(Range.Constant(1, i)))
+                select Assert.True(j <= i));
+        }
+
+        [Fact]
+        public void CanDependOnEarlierValuesWithBool()
+        {
+            Check(
+                from i in ForAll(Gen.Int32(Range.Constant(1, 10)))
+                from j in ForAll(Gen.Int32(Range.Constant(1, i)))
+                select j <= i);
+        }
+
+        //[Fact]
+        //public void ShrinkTest()
+        //{
+        //    Check(
+        //        from s in ForAll(Gen.String(Range.Constant(1, 100), Gen.CharAlpha))
+        //        from s2 in ForAll(Gen.String(Range.Constant(1, 10), Gen.CharAlpha))
+        //        select Assert.DoesNotContain(s2, s));
+        //}
+
+        //[Fact]
+        //public void ShrinkTestt()
+        //{
+        //    Check(
+        //        from s in ForAll(Gen.String(Range.Constant(1, 100), Gen.CharAlpha))
+        //        from s2 in ForAll(Gen.String(Range.Constant(1, 10), Gen.CharAlpha))
+        //        select !s.Contains(s2));
+        //}
```

Comment:
Maybe we should remove this commented-out blocks of code and just add some `FIXME`s?

The problem here is that these might get stale, and after some time we might not know what to do with them; and they might end up staying there, commented-out, forever.